crates/wasmtime/src/runtime/store.rs

```diff
@@ -1333,30 +1383,67 @@ impl StoreOpaque {
             .map(|memory| unsafe { Memory::from_wasmtime_memory(memory, self) })
     }
 
+    /// Iterate over all tables (host- or Wasm-defined) within this store.
+    pub fn for_each_table<'a>(&'a mut self, mut f: impl FnMut(&mut Self, Table)) {
+        // NB: Host-created tables have dummy instances. Therefore, we can get
+        // all memories in the store by iterating over all instances (including
+        // dummy instances) and getting each of their defined memories.
+
+        let mut instances = mem::take(&mut self.instances);
+
+        let result = std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
```

Comment:
If this `catch_unwind` is only here to restore `instances`, can this be done with a `Drop` rather than catching unwind?